cookbooks/bcpc/attributes/default.rb

```diff
@@ -3,101 +3,163 @@
 #  General configuration for this cluster
 #
 ###########################################
-default['bcpc']['country'] = "US"
-default['bcpc']['state'] = "NY"
-default['bcpc']['location'] = "New York"
-default['bcpc']['organization'] = "Bloomberg"
+default['bcpc']['country'] = 'US'
+default['bcpc']['state'] = 'NY'
+default['bcpc']['location'] = 'New York'
+default['bcpc']['organization'] = 'Bloomberg'
 # Should be kvm (or qemu if testing in VMs)
-default['bcpc']['virt_type'] = "kvm"
+default['bcpc']['virt_type'] = 'kvm'
 # Region name for this cluster
 default['bcpc']['region_name'] = node.chef_environment
 # Domain name that will be used for DNS
-default['bcpc']['domain_name'] = "bcpc.example.com"
+default['bcpc']['domain_name'] = 'bcpc.example.com'
 # Key if Cobalt+VMS is to be used
 default['bcpc']['vms_key'] = nil
 
 default['bcpc']['encrypt_data_bag'] = false
 
-# Specify the kernel you wish to install. For default latest LTS kernel use "linux-server"
-default['bcpc']['bootstrap']['preseed']['kernel'] = "linux-generic-lts-trusty"
-default['bcpc']['bootstrap']['preseed']['add_kernel_opts'] = "console=ttyS0"
-default['bcpc']['bootstrap']['preseed']['late_command'] = "true"
+# Specify the kernel you wish to install.
+# For default, (latest LTS kernel) use "linux-server"
+default['bcpc']['bootstrap']['preseed']['kernel'] = 'linux-generic-lts-trusty'
+default['bcpc']['bootstrap']['preseed']['add_kernel_opts'] = 'console=ttyS0'
+default['bcpc']['bootstrap']['preseed']['late_command'] = 'true'
 default['bcpc']['bootstrap']['admin_users'] = []
 
+#
+# The node_number is used to derive Kafka broker IDs, Zookeeper myid
+# files, keepalived node priorities, and other values.  It must be
+# unique within a cluster.
+#
+# The node number is generated from the integer value of the
+# default interface mac_address, modulo Java's Integer.MAX_VALUE.
+#
+# For 8-bit node numbers, (e.g. zookeeper, keepalived) there is a
+# helper method defined in libraries/node_numbers.rb
+#
+unless node['bcpc']['node_number']
```

Comment:
On general principal, doing this in an attribute file and not in a function is pretty concerning to me for maintainability -- it is quite a bit akin to a side-affect not actual intention from my viewpoint.